File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
import java.util.*;
class Solution {
    public int lengthOfLongestSubstring(String s) {
      int i=0,j=0,length=0;
      ArrayList<Character> list=new ArrayList<>();
      while(j<s.length()){
        if(!list.contains(s.charAt(j))){
            list.add(s.charAt(j));
            j++;
            length=Math.max(length,list.size());
        }
        else{
            list.remove(Character.valueOf(s.charAt(i)));
            i++;
        }
      }
      return length;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        r=[]
        l=0
        for i in range(len(s)):
            if s[i] not in r:
                r.append(s[i])
            else:
                l=max(len(r),l)
                while s[i] in r:
                    r.pop(0)
                r.append(s[i])
        return max(len(r),l)